### PR TITLE
Changes drone to remove the val isConnected in DroneServiceImpl.kt

### PR DESCRIPTION
**Pull request by Anghul**
I changed the implementation of the functions connecting and disconnecting the drone to remove the need for the var "isConnected"

### CODE REVIEW

**Comment by GabrielFleischer** [comment]
I think we should keep the current drone instance in the service itself (DroneSreviceImpl) and for the tests, create a DroneInstanceFactory providing the System with somthing like createSimulation(ip, port) and createDrone().

So that we can provide a mocked factory to handle the tests.

File: app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServerFactoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class DroneServerFactoryImpl: DroneServerFactory {
+
+    override fun createSimulation(simIP: String, simPort: String): DroneServerFactory.InstanceContainer {
+        val mavsdkServer = MavsdkServer()
+        val cloudSimIPAndPort = "tcp://${simIP}:${simPort}"
+        val mavsdkServerPort = mavsdkServer.run(cloudSimIPAndPort)
+        return DroneServerFactory.InstanceContainer(mavsdkServer, System("localhost", mavsdkServerPort))
+    }
+
+    override fun createDrone(): DroneServerFactory.InstanceContainer {
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServerFactoryImpl.kt:20]: Does this really provide a correct drone instance?

File: app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServiceImpl.kt [hunk 2]
```diff
@@ -11,30 +11,31 @@ package ch.epfl.sdp.drone3d.drone
...
-            val mavsdkServerPort = mavsdkServer.run(cloudSimIPAndPort)
-            System("localhost", mavsdkServerPort)
-        } else {
-            val mavsdkServer = MavsdkServer()
-            val mavsdkServerPort = mavsdkServer.run()
-            System("localhost", mavsdkServerPort)
-        }
-    }
+    private var simIP = DEFAULT_IP
+    private var simPort = DEFAULT_IP
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServiceImpl.kt:26]: Those 2 equalities seem odd as having the same value

File: app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServerFactoryImpl.kt (hunk 1 above)
Comment (GabrielFleischer) [reply app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServerFactoryImpl.kt:20]: Yes, from my understanding the instance is correct until the server is stopped

File: app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServiceImpl.kt (hunk 2 above)
Comment (Anghul) [reply app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServiceImpl.kt]: It was an error and has been corrected

Comment (GabrielFleischer) [reply app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServiceImpl.kt]: true, that's a bad copy/paste

**Review by Squalene — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneDataImpl.kt [hunk 3]
```diff
@@ -52,42 +47,63 @@ class DroneDataImpl constructor(val provider: DroneService) : DroneData {
     }
 
     private fun createDefaultSubs() {
```
Comment (loh0um) [inline app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneDataImpl.kt:49]: Some documentation of the methods might make the reading of the code easier

**Review by GabrielFleischer — APPROVED** [review]

**Review by loh0um — APPROVED** [review]

**Pull request merged**